src/ipvs/ip_vs_proto_udp.c

```diff
@@ -327,20 +440,42 @@ static int udp_fnat_in_handler(struct dp_vs_proto *proto,
                     struct dp_vs_conn *conn,
                     struct rte_mbuf *mbuf)
 {
-    struct udp_hdr *uh;
+    struct udp_hdr *uh = NULL;
+    struct iphdr *iph = (void *)ip4_hdr(mbuf);
+    struct opphdr *opp = NULL;
 
     /* cannot use mbuf_header_pointer() */
     if (unlikely(mbuf->data_len < ip4_hdrlen(mbuf) + sizeof(struct udp_hdr)))
         return EDPVS_INVPKT;
-    uh = rte_pktmbuf_mtod_offset(mbuf, struct udp_hdr *, ip4_hdrlen(mbuf));
+
+    if (iph->protocol == IPPROTO_UDP) {
+        uh = (void *)iph + ip4_hdrlen(mbuf);
+    } else if (iph->protocol == IPPROTO_OPT) {
+        opp = (void *)iph + ip4_hdrlen(mbuf);
+
+        uh = (void *)opp + ntohs(opp->length);
```

Comment:
1. `void *` is *one byte offset* , same with `char *` and,
2. `void *` can be translated to any types.

Otherwise, if use `char *`, the code may looks like,
` uh = (struct udphdr *)((char *)opp + ...)`

Reply:
ok